src/NuGet.Core/NuGet.Packaging/Signing/Archive/SignedPackageArchiveUtility.cs

```diff
@@ -88,6 +88,33 @@ private static bool IsSigned(BinaryReader reader)
             return false;
         }
 
+        /// <summary>
+        /// Signs a Zip with the contents in the SignatureStream using the writer.
+        /// The reader is used to read the exisiting contents for the Zip.
+        /// </summary>
+        /// <param name="signatureStream">MemoryStream of the signature to be inserted into the zip.</param>
+        /// <param name="reader">BinaryReader to be used to read the existing zip data.</param>
+        /// <param name="writer">BinaryWriter to be used to write the signature into the zip.</param>
+        public static void SignZip(MemoryStream signatureStream, BinaryReader reader, BinaryWriter writer)
+        {
+            if (signatureStream == null)
+            {
+                throw new ArgumentNullException(nameof(signatureStream));
+            }
+
+            if (reader == null)
+            {
+                throw new ArgumentNullException(nameof(reader));
+            }
+
+            if (writer == null)
+            {
+                throw new ArgumentNullException(nameof(writer));
+            }
+
+            SignedPackageArchiveIOUtility.WriteSignatureIntoZip(signatureStream, reader, writer);
```

Comment:
lower level methods are all internal/private. it makes more sense to have the checks in a public method. Why do you think it would be beneficial there?

Reply:
I think even though it is internal, it would be better to do it there so that every caller doesn't have to do these checks manually. It could be a refactor for later, so if you have a preference then I'm okay.